crates/fuel-core/src/schema/coin.rs

```diff
@@ -90,13 +94,12 @@ impl CoinQuery {
         ctx: &Context<'_>,
         #[graphql(desc = "The ID of the coin")] utxo_id: UtxoId,
     ) -> async_graphql::Result<Option<Coin>> {
-        let utxo_id = utxo_id.0;
-        let db = ctx.data_unchecked::<Database>().clone();
-        let block = db
-            .storage::<Coins>()
-            .get(&utxo_id)?
-            .map(|coin| Coin(utxo_id, coin.into_owned()));
-        Ok(block)
+        let data = CoinQueryContext(ctx.data_unchecked());
+        let coin = data.coin(utxo_id.0)?;
+
+        let coin = coin.map(|coin| Coin(utxo_id.0, coin));
```

Comment:
Maybe better to implement `From<fuel_core_types::entities::coin::Coin>` and use `into`.